File: src/num_fields.f
c********************************************************************
        integer function num_fields(ascii_line)
c
c	Determines the number of separate fields in an ascii line.
c	line_len is no longer used, and was left into the call
c	for compatability with
c	older codes, but has been removed. Hopefully I found all the other 
c	spots it has been called from in the codes!
c 
        character ascii_line*(*)
	character*1 tab
        logical new
 
        itemp=0
        new=.true.
	tab=char(9)
 
        do index=1,len(ascii_line) 
                if(ascii_line(index:index).eq.' '.or.
     &             ascii_line(index:index).eq.','.or.
     &             ascii_line(index:index).eq.tab) then
                        new=.true.
                else
                        if(new) itemp=itemp+1
                        new=.false.
                endif
 
        enddo
 
        num_fields=itemp
        return
        end

File: py/num_fields.py
def num_fields(ascii_line):
    """
    Determines the number of separate fields in an ASCII line.
    Fields are separated by spaces, commas, or tabs.

    Parameters:
    ascii_line (str): The input ASCII line as a string.

    Returns:
    int: The number of separate fields in the line.
    """
    itemp = 0
    new = True  # Indicates if the previous character was a delimiter
    delimiters = {' ', ',', '\t'}  # Set of delimiter characters

    for char in ascii_line:
        if char in delimiters:
            new = True
        else:
            if new:
                itemp += 1  # Increment field count at the start of a new field
            new = False  # We are inside a field

    return itemp
